### PR TITLE
feat: adds support ReferencesMany relation

### PR DESCRIPTION
**Pull request by mikeevstropov**
Definition in Model.
```typescript
@referencesMany(() => Account)
accountIds: string[]; // or number[]
```
Include in query filter.
```
{"include": ["accounts"]}
```
Response will contain resolved relations in `accounts` field. E.g.
```json
{
  "id": "customer-1",
  "accountIds": [
    "account-1",
    "account-2"
  ],
  "accounts": [
    {"id": "account-1", "balance": 10},
    {"id": "account-2", "balance": 20},
  ]
}
```

Related
https://github.com/loopbackio/loopback-next/issues/2488
https://github.com/loopbackio/loopback-next/issues/1450
https://github.com/loopbackio/loopback-next/issues/1920

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [x] Affected example projects in `examples/*` were updated

### CODE REVIEW

**Comment by mikeevstropov** [comment]
Done

**Comment by edgency** [comment]
Looks like no one needs this add-on, sadness

**Comment by agnes512** [comment]
https://github.com/loopbackio/loopback-next/pull/8271

File: examples/references-many/src/__tests__/acceptance/customer.acceptance.ts
Comment (achrinza) [inline examples/references-many/src/__tests__/acceptance/customer.acceptance.ts:1]: Copyright year should be `2022` for this package.

**Review by achrinza — APPROVED** [review]
Generally LGTM :+1: Will see if any of the other maintainers have any objection.

**Comment by mikeevstropov** [comment]
Redundant property `itemType` has been removed from the relation decorator. Example updated.

**Review by raymondfeng — APPROVED** [review]

**Comment by dhmlau** [comment]
@mikeevstropov, your PR is almost good to go. Could you please squash the commits, esp the ones with `Merge branch master...`? Thanks.

cc @achrinza

**Comment by mikeevstropov** [comment]
@dhmlau Hi. I'm not sure if I did it right, just squashed the commits into one.

**Comment by achrinza** [comment]
@mikeevstropov If the changes are only localised in the `@loopback/repository` package, you can select only that specific set of files:

```sh
gitCommitHash=$(git log -1 -pretty=%H)
# Take note of the hash in case you need to cherry-pick from reflog:
echo "$gitCommitHash"
git reset --hard master
git checkout "$gitCommitHash" -- packages/repository/src
```

The above commands _should_ cherry-pick all of the files needed. Please confirm everything is in order before pushing it as this is a destructive action. You can also choose to checkout these changes to a new branch and create a new PR.

If you're not comfortable with this, the maintainers can help make the necessary changes from our end.

**Comment by achrinza** [comment]
https://github.com/loopbackio/loopback-next/pull/8455 is my attempt at "recovering" the original PR using the steps above. If it looks correct, you can try the steps on this PR, or we can use that new PR if that's preferred.

**Comment by mikeevstropov** [comment]
@achrinza The commit is renamed as you suggest.

**Comment by achrinza** [comment]
Thanks for your contributions, @mikeevstropov! They have been merged 🎉

**Pull request merged**
